packages/back-end/src/util/organization.util.ts

```diff
@@ -50,6 +51,43 @@ export const ALL_PERMISSIONS = [
   ...ENV_SCOPED_PERMISSIONS,
 ];
 
+export function getUserPermissions(user: any, org: OrganizationInterface) {
+  const roles = getRoles(org);
+  const globalRolePermissions = roles.find((r) => r.id === user.role);
+  const userPermissions: any = {};
+
+  userPermissions.global = {
+    environments: user.environments || [],
+    limitAccessByEnvironment: user.limitAccessByEnvironment || false,
+    permissions: {},
+  };
+
+  ALL_PERMISSIONS.forEach((permission) => {
+    userPermissions.global.permissions[
+      permission
+    ] = globalRolePermissions?.permissions.includes(permission) ? true : false;
+  });
+
+  userPermissions.projects = {};
+
+  user.projectRoles.forEach((projectRole: any) => {
+    const projectRolePermissions = roles.find((r) => r.id === projectRole.role);
+    userPermissions.projects[projectRole.project] = {
+      limitAccessByEnvironment: projectRole.limitAccessByEnvironment || false,
+      environments: projectRole.environments || [],
+      permissions: {},
+    };
+    ALL_PERMISSIONS.forEach((permission) => {
+      userPermissions.projects[projectRole.project].permissions[
```

Comment:
TODO: We're looping through each project and then looping through all permissions - is there a more performant way to do this?